encoder/src/main/java/com/pedro/encoder/utils/gl/SizeCalculator.java

```diff
@@ -36,12 +36,16 @@ public static void calculateViewPort(boolean keepAspectRatio, int mode, int prev
   }
 
   public static void updateMatrix(int rotation, int width, int height, boolean isPreview,
-      boolean isPortrait, float[] MVPMatrix) {
+                                  boolean isPortrait, boolean isCameraFront,
+                                  boolean flipStreamHorizontal, boolean flipStreamVertical, float[] MVPMatrix) {
     Matrix.setIdentityM(MVPMatrix, 0);
     PointF scale = getScale(rotation, width, height, isPortrait, isPreview);
     Matrix.scaleM(MVPMatrix, 0, scale.x, scale.y, 1f);
     if (!isPreview && !isPortrait) rotation += 90;
     Matrix.rotateM(MVPMatrix, 0, rotation, 0f, 0f, -1f);
+    if (!isPreview && isCameraFront) {
```

Comment:
I think that user should decide whether he wants to flip front or back camera

Reply:
I think that `isCameraFront` param is not needed and user should decide whether to flip it.
So basically you don't need a callback for camera at all, or you could move it to Camera2ApiManager

so on theuser side it would be like 
```
fun switchCamera {
    try {
       livestream.switchCamera()
   } finally {
       liveStream.glInterface.flipStreamVertical = livestream.isFrontCamera
   }
}
```

or 
```
livstream.cameraChangeListener() { isFrontCamera ->
   liveStream.glInterface.flipStreamVertical = isFrontCamera
}

```

Reply:
Applied

Reply:
However, I'm not sure if putting the flip in `finally` block is a good idea. After all, it's executed after `catch` as well.